File: dev-docs/snippets/documents/list-access-requests.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List pending access requests for a document (owner/admin only).
export async function listAccessRequests(
  client: JsBaoClient,
  documentId: string,
) {
  // #region example
  const requests = await client.documents.listAccessRequests(documentId);
  // #endregion example
  return requests;
}

File: dev-docs/snippets/documents/list-access-requests.swift
import JsBaoClient

// List pending access requests for a document (owner/admin only) as typed
// `[DocumentAccessRequest]`.
func listAccessRequests(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let requests = try await client.documents.listAccessRequests(
    documentId: documentId
  )
  let firstRequester = requests.first?.requesterId
  // #endregion example
  _ = firstRequester
}
